tests/plugins/copyformatting/undointegration.js

```diff
@@ -0,0 +1,83 @@
+/* bender-tags: copyformatting */
+/* bender-ckeditor-plugins: wysiwygarea, toolbar, copyformatting, undo */
+/* bender-include: _helpers/tools.js*/
+/* bender-ui: collapsed */
+
+( function() {
+	'use strict';
+
+	var leftMouseButton = CKEDITOR.env.ie && CKEDITOR.env.version < 9 ? 1 : CKEDITOR.MOUSE_BUTTON_LEFT;
+
+	bender.editor = {
+		config: {
+			allowedContent: true
+		}
+	};
+
+
+	bender.test( {
+		// (#2780)
+		'test basic undo integration': function() {
+			var editor = this.editor,
+				bot = this.editorBot,
+				sel = editor.getSelection(),
+				el1,
+				el2;
+
+			resetEditorStatusAndCheckBasicAsserts( bot );
+
+			el1 = editor.editable().findOne( '#one' );
+			el2 = editor.editable().findOne( '#two' );
+
+			for ( var i = 0; i < editor.undoManager.limit; i++ ) {
+				sel.selectElement( i % 2 ? el1 : el2 );
+				editor.document.fire( 'mouseup', new CKEDITOR.dom.event( {
+					button: leftMouseButton,
+					target: editor.editable()
+				} ) );
+			}
+
+			assert.areSame( 1, editor.undoManager.index, 'There shouldn\'t be new undo steps and editor should remain on the 1st step.' );
+			assert.isTrue( editor.undoManager.undoable(), 'Editor should have a possibility to undo.' );
+		},
+		// (#2780)
+		'test basic redo integration': function() {
+			var editor = this.editor,
+				bot = this.editorBot,
+				sel = editor.getSelection();
+
+			resetEditorStatusAndCheckBasicAsserts( bot );
+
+			bot.execCommand( 'undo' );
+
+			sel.selectElement( editor.editable().findOne( '#one' ) );
+			editor.document.fire( 'mouseup', new CKEDITOR.dom.event( {
+				button: leftMouseButton,
+				target: editor.editable()
+			} ) );
+
+			assert.isTrue( editor.undoManager.redoable(), 'Editor should has possibility to redo.' );
+		}
+	} );
+
+	function resetEditorStatusAndCheckBasicAsserts( bot ) {
```

Comment:
Maybe we can simplify the name, e.g. `testBasicIntegration`?

Reply:
Actually this helper function seems more like a basic undo integration test. Such things should be sufficiently tested by our undo plugin tests. Resetting `undoManager` and creating the first snapshot is basically all we need here.